pinot-segment-local/src/main/java/org/apache/pinot/segment/local/realtime/impl/invertedindex/RealtimeLuceneTextIndex.java

```diff
@@ -119,7 +122,12 @@ public MutableRoaringBitmap getDocIds(String searchQuery) {
     Callable<MutableRoaringBitmap> searchCallable = () -> {
       IndexSearcher indexSearcher = null;
       try {
-        Query query = new QueryParser(_column, _analyzer).parse(searchQuery);
+        QueryParser parser = new QueryParser(_column, _analyzer);
+        parser.setAllowLeadingWildcard(true);
```

Comment:
Could setting `parser.setAllowLeadingWildcard(true);` change behavior for all queries?

If so we should do this only when `_enablePrefixSuffixMatchingInPhraseQueries` is set.